solr-ingest/src/main/java/edu/unc/lib/dl/data/ingest/solr/filter/SetDatastreamFilter.java

```diff
@@ -60,22 +60,20 @@ public void filter(DocumentIndexingPackage dip) throws IndexingException {
         ContentObject contentObj = dip.getContentObject();
 
         FileObject fileObj = getFileObject(contentObj);
-        if (fileObj == null) {
-            return;
+        if (fileObj != null) {
```

Comment:
No datastreams are going to get indexed for objects other than works and files, i think? Did you need to get rid of the other side of this conditional? We still need to index other objects, we just don't need to restrict it to CollectionObjects, which is what you had before.

Reply:
So are we back to this?
`   if (fileObj != null) {
            boolean ownedByOtherObject = contentObj instanceof WorkObject;
            // Retrieve list of datastreams associated with this object
            List<Datastream> datastreams = getDatastreams(fileObj, ownedByOtherObject);
            // Retrieve list of derivatives associated with the object
            List<Datastream> derivatives = getDerivatives(fileObj.getPid(), ownedByOtherObject);
            datastreams.addAll(derivatives);

            IndexDocumentBean doc = dip.getDocument();

            doc.setDatastream(getDatastreamStrings(datastreams));
            doc.setFilesizeTotal(getFilesizeTotal(datastreams));
            doc.setFilesizeSort(getFilesize(datastreams));
        } else {
            List<Datastream> derivatives = getDerivatives(contentObj.getPid(), false);
            List<Datastream> datastreams = new ArrayList<>(derivatives);

            IndexDocumentBean doc = dip.getDocument();

            doc.setDatastream(getDatastreamStrings(datastreams));
        }`

Reply:
yes almost, my last comment was that in the second block, `derivatives` and `datastreams` have the exact same stuff in them, so there's no reason to make the second variable. We do need the `else` block which covers objects besides works and files.